.github/workflows/ci.yml

```diff
@@ -97,6 +97,47 @@ jobs:
           bazel test //tensorboard/compat/tensorflow_stub:gfile_s3_test &&
           bazel test //tensorboard/summary/writer:event_file_writer_s3_test
 
+  build-data-server-pip:
+    runs-on: ${{ matrix.platform }}
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: ['ubuntu-16.04']
+        rust_version: ['1.48.0']
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-python@v1
+        with:
+          python-version: '3.8'
+          architecture: 'x64'
+      - name: 'Install Rust toolchain'
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ matrix.rust_version }}
+      - name: 'Install Python packaging deps'
+        run: |
+          python -m pip install -U pip
+          pip install \
+            -c ./tensorboard/pip_package/requirements.txt \
+            -c ./tensorboard/pip_package/requirements_dev.txt \
+            setuptools wheel
+      - name: 'Build'
+        run: cd tensorboard/data/server/ && cargo build --release
+      - name: 'Test'
+        run: cd tensorboard/data/server/ && cargo test --release
+      - name: 'Package'
+        run: |
+          rm -rf /tmp/pip_package && mkdir /tmp/pip_package
+          python tensorboard/data/server/pip_package/build.py \
```

Comment:
Just wondering, you do have a [bazel target for this](https://github.com/tensorflow/tensorboard/pull/4429/files#diff-cd744ef136a1ace4eea485e8399ac9f8b4aa457ba39334f7c0bf04daa8fbd2aaR6-R13), right? Are you manually invoking it here to (1) not require bazel and (2) specify output like code here?

Reply:
Exactly. Not requiring Bazel is the main point. We can build the whole
Rust package from scratch faster `bazel fetch //tensorboard/...` with
cache, and by building with Cargo rather than Bazel we avoid needing to
get Bazel set up on macOS and Windows CI (with any caching headaches
that that might entail).